campaigns/cm_time.go

```diff
@@ -0,0 +1,27 @@
+package campaigns
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+)
+
+type CmTime time.Time
```

Comment:
I like the idea of introducing a custom marshaller, but unfortunately time values are not formatted consistently across CM API. 😞 Check [this](https://www.campaignmonitor.com/api/transactional/#smart-email-listing) out for example. It's definitely possible to use CMTime whenever possible (which is maybe more than 90% of the API to be fair), but it will make the wrapper inconsistent. I strongly believe we should go everywhere CMTime, or no CMTime at all. An intermediate approach would be to move CMTime into the `internal` package and not expose it. The benefit would be that the time serialisation/de-serialisation will be centralised in one place, but we will still need those intermediate contracts for sure.